src/test/Microsoft.DotNet.CoreSetup.Packaging.Tests/NuGetArtifactTester.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.CoreSetup.Test;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Xml.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Packaging.Tests
+{
+    public class NuGetArtifactTester : IDisposable
+    {
+        public static NuGetArtifactTester Open(
+            RepoDirectoriesProvider dirs,
+            string project,
+            string id = null)
+        {
+            var tester = OpenOrNull(dirs, project, id);
+            Assert.NotNull(tester);
+            return tester;
+        }
+
+        public static NuGetArtifactTester OpenOrNull(
+            RepoDirectoriesProvider dirs,
+            string project,
+            string id = null)
+        {
+            id = id ?? project;
+
+            string nuspecPath = Path.Combine(
+                dirs.BaseBinFolder,
+                project,
+                dirs.Configuration,
+                "specs",
+                $"{id}.nuspec");
+
+            if (!File.Exists(nuspecPath))
+            {
+                return null;
+            }
+
+            PackageIdentity builtIdentity = new NuspecReader(nuspecPath).GetIdentity();
+
+            string path = Path.Combine(
+                dirs.BaseArtifactsFolder,
+                "packages",
+                dirs.Configuration,
+                "Shipping",
+                $"{builtIdentity}.nupkg");
+
+            // If the nuspec exists, the nupkg should exist.
+            Assert.True(File.Exists(path));
+
+            return new NuGetArtifactTester(path);
+        }
+
+        private readonly PackageArchiveReader _reader;
+
+        public NuGetArtifactTester(string file)
+        {
+            _reader = new PackageArchiveReader(ZipFile.Open(file, ZipArchiveMode.Read));
+        }
+
+        public void Dispose()
+        {
+            _reader.Dispose();
+        }
+
+        public void IsTargetingPack()
+        {
+            IsFrameworkPack();
+
+            Assert.NotEmpty(_reader.GetFiles("ref"));
+            Assert.Empty(_reader.GetFiles("runtimes").Concat(_reader.GetFiles("lib")));
```

Comment:
It seems like it would be easier to track down issues if the asserts were more fine grained.  Why not break out `Assert.Empty(_reader.GetFiles("lib"));`

Reply:
My reasoning was that combining them reduces the onion peeling if both runtime and lib files started showing up. I believe the assert shows the strings, so it's diagnosable fine. Changing, though--splitting this up also makes it more readable IMO and it's unlikely this is a practical issue.

Reply:
> I believe the assert shows the strings

Not true!